### ISSUE TITLE
Health Check Call hangs if invoked with certain Http Status Codes (weird ones)

### ISSUE THREAD

**Issue by rajanadar** [hashicorp/vault#1849]
this might be a minor issue, since it involves folks customizing the api will funky status codes. but here it is anyways.

Version: 0.6.1

1. have vault not initialized.
2. Try to hit the Vault health url: http://127.0.0.1:8200/v1/sys/health?uninitcode=600
The call just hangs, and doesn't return.

3. Also, try hitting http://127.0.0.1:8200/v1/sys/health?uninitcode=100
the call returns with 100 status, but with an empty response.

basically the api works if the custom code is between 200 and 599, inclusive.
applies to all custom codes.. uninit, seal, active, standby etc.

also, wanted to know the thinking behind customizing the codes. is it to have load balancers deal with sealed/uninitialized vaults, as healthy as well, if need be; but by default treat them as unavailable?

**Comment by jefferai** [comment]
I can't replicate this:
```
$ curl 'http://127.0.0.1:8200/v1/sys/health?uninitcode=100'
{"initialized":true,"sealed":false,"standby":false,"server_time_utc":1472946331,"version":"0.6.2-dev","cluster_name":"vault-cluster-10fef3e6","cluster_id":"d2d112cd-c270-ed0f-9df8-2870a6cb80c6"}

$ curl 'http://127.0.0.1:8200/v1/sys/health?uninitcode=600'
{"initialized":true,"sealed":false,"standby":false,"server_time_utc":1472946336,"version":"0.6.2-dev","cluster_name":"vault-cluster-10fef3e6","cluster_id":"d2d112cd-c270-ed0f-9df8-2870a6cb80c6"}
```
What version of Vault are you using?

**Comment by jefferai** [comment]
Ah, nm. I forgot that you said that it was an uninitialized vault. Can replicate.

**Comment by jefferai** [comment]
Ah. I know what this is :-)

The reason is that 100 is a reserved code. `curl` is expecting more data. Things are behaving appropriately, it's just not an appropriate status code to use!
